app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/returns/GlarCalculatorEligibilityUtil.scala

```diff
@@ -0,0 +1,149 @@
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.AssetType.{NonResidential, Residential}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.CompleteReturn.CompleteSingleDisposalReturn
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.DraftSingleDisposalReturn
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.IndividualUserType.Self
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.ReliefDetailsAnswers.CompleteReliefDetailsAnswers
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.SingleDisposalTriageAnswers.CompleteSingleDisposalTriageAnswers
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+
+import scala.concurrent.{ExecutionContext, Future}
+@ImplementedBy(classOf[GlarCalculatorEligibilityUtilImpl])
+trait GlarCalculatorEligibilityUtil {
+
+  def isEligibleForFurtherReturnOrAmendCalculation(fillingOutReturn: FillingOutReturn)(implicit
+    hc: HeaderCarrier,
+    request: RequestWithSessionData[_]
+  ): EitherT[Future, Error, FurtherReturnEligibility]
+
+}
+
+final case class FurtherReturnEligibility(isEligible: Boolean, previousReturnsImplyEligibility: Option[Boolean])
+
+@Singleton
+class GlarCalculatorEligibilityUtilImpl @Inject() (
+  returnsService: ReturnsService,
+  servicesConfig: ServicesConfig
+)(implicit ec: ExecutionContext)
+    extends GlarCalculatorEligibilityUtil
+    with SessionUpdates {
+
+  val amendAndFurtherReturnCalculationsEnabled =
+    servicesConfig.getBoolean("amend-and-further-returns-calculator.enabled")
+
+  def isEligibleForFurtherReturnOrAmendCalculation(
+    fillingOutReturn: FillingOutReturn
+  )(implicit
+    headerCarrier: HeaderCarrier,
+    request: RequestWithSessionData[_]
+  ): EitherT[Future, Error, FurtherReturnEligibility] = {
+
+    def isEligible(
+      fillingOutReturn: FillingOutReturn,
+      completeSingleDisposalTriageAnswers: CompleteSingleDisposalTriageAnswers,
+      reliefDetailsAnswers: CompleteReliefDetailsAnswers
+    ): EitherT[Future, Error, FurtherReturnEligibility] = {
+      val eligibleAssetType =
+        completeSingleDisposalTriageAnswers.assetType === Residential || completeSingleDisposalTriageAnswers.assetType === NonResidential
+
+      val noOtherReliefs = reliefDetailsAnswers.otherReliefs.isEmpty
+
+      val underPreviousReturnLimit = fillingOutReturn.previousSentReturns.exists(_.summaries.length <= 10)
+
+      val isSelf = completeSingleDisposalTriageAnswers.individualUserType.contains(Self)
+
+      val currentReturnIsEligible = eligibleAssetType && noOtherReliefs && underPreviousReturnLimit && isSelf
+
+      if (
+        fillingOutReturn.previousReturnsImplyEligilityForFurtherReturnCalculation.contains(
+          true
+        ) && currentReturnIsEligible
+      ) {
+        EitherT.right(Future(FurtherReturnEligibility(true, Some(true))))
+      } else if (currentReturnIsEligible) {
```

Comment:
this branch will get triggered if `previousReturnsImplyEligilityForFurtherReturnCalculation` is `false`. The correct logic would be something like:
```scala
fillingOutReturn.previousReturnsImplyEligilityForFurtherReturnCalculation match {
  case Some(previousReturnsImplyEligilityForFurtherReturnCalculation) =>
    EitherT.pure(
       if(previousReturnsImplyEligilityForFurtherReturnCalculation) 
          FurtherReturnEligibility(currentReturnIsEligible, Some(true))
       else
           FurtherReturnEligibility(false, Some(true))
   )

   case None =>
     if(!currentReturnIsEligible)
       EitherT.pure(FurtherReturnEligibility(false, None)))
     else { ... } 
```
Also, note that you can use `EitherT.pure` instead of `.right` to avoid having to explicitly create a `Future`